### PR TITLE
feat: multiselect connection while issuance

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What ?
- Implemented multi-select connections functionality while issuance

### Why?
- Implementing multi-select connections functionality during issuance enhances efficiency by streamlining the process for users to issue credentials to multiple recipients simultaneously. 
- This feature improves user experience and reduces the time and effort required for issuing credentials individually to each connection.

### CODE REVIEW

File: apps/api-gateway/src/issuance/dtos/issuance.dto.ts [hunk 1]
```diff
@@ -121,18 +121,15 @@ class Attribute {
...
-    @IsNotEmpty({ message: 'isRequired property is required' })
     isRequired?: boolean = false;
 
 }
-
-class CredentialsIssuanceDto {
+export class CredentialsIssuanceDto {
     @ApiProperty({ example: 'string' })
-    @IsNotEmpty({ message: 'Please provide valid credential definition id' })
+    @IsNotEmpty({ message: 'credentialDefinitionId is required' })
```
Comment (tipusinghaw) [inline apps/api-gateway/src/issuance/dtos/issuance.dto.ts:129]:
```suggestion
    @IsNotEmpty({ message: 'Credential definition Id is required' })
```

File: apps/api-gateway/src/issuance/issuance.controller.ts [hunk 2]
```diff
@@ -63,6 +62,7 @@ import { RpcException } from '@nestjs/microservices';
 /* eslint-disable @typescript-eslint/no-unused-vars */
 import { user } from '@prisma/client';
 import { IGetAllIssuedCredentialsDto } from './dtos/get-all-issued-credentials.dto';
+import { IssueCredentialDto } from './dtos/multiConnection.dto';
```
Comment (tipusinghaw) [inline apps/api-gateway/src/issuance/issuance.controller.ts:65]:
```suggestion
import { IssueCredentialDto } from './dtos/multi-connection.dto';
```

File: apps/connection/src/connection.repository.ts [hunk 3]
```diff
@@ -149,7 +149,7 @@ export class ConnectionRepository {
           break;
       }
 
-      const agentDetails = await this.prisma.connections.upsert({
+      const connectionDetails = await this.prisma.connections.upsert({
```
Comment (tipusinghaw) [inline apps/connection/src/connection.repository.ts:152]: it's good the change the unrelated veriable names

File: apps/api-gateway/src/issuance/dtos/issuance.dto.ts (hunk 1 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/issuance/dtos/issuance.dto.ts]: Resolved

File: apps/api-gateway/src/issuance/issuance.controller.ts (hunk 2 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/issuance/issuance.controller.ts]: Resolved

File: apps/connection/src/connection.repository.ts (hunk 3 above)
Comment (bhavanakarwade) [reply apps/connection/src/connection.repository.ts:152]: Yes

Comment (KulkarniShashank) [inline apps/connection/src/connection.repository.ts:152]: You can directly return the result without assigning it to a variable:

```suggestion
      return this.prisma.connections.upsert({
```

Comment (bhavanakarwade) [reply apps/connection/src/connection.repository.ts]: Resolved

**Review by KulkarniShashank — APPROVED** [review]
LG

**Review by tipusinghaw — APPROVED** [review]

**Pull request merged**